File: find-pivot-index/find-pivot-index.py
class Solution(object):
    def pivotIndex(self, nums):
        sum_left = 0
        sum_right = sum(nums)
        
        for i in range(len(nums)):
            sum_left += nums[i]
            if sum_left == sum_right:
                return i 
            sum_right -= nums[i]
            
            
        return -1 

File: find-pivot-index/find-pivot-index.js
var pivotIndex = function (nums) {
    left_sum = 0;
    right_sum = 0;
    for (let i = 0; i < nums.length; i++) {
        right_sum += nums[i]
    }

    for (let i = 0; i < nums.length; i++) {
        left_sum += nums[i]
        if (left_sum == right_sum) {
            return i
        }
        right_sum -= nums[i]
    }

    return -1
};